api/image.go

```diff
@@ -243,12 +253,11 @@ func (api *API) doUpdateImage(w http.ResponseWriter, req *http.Request, id strin
 			return nil
 		}
 	}
-	updatedImage.Refresh()
 	return updatedImage
 }
 
-// UploadImageHandler updates the image with the provided upload path and sends a kafka 'image uploaded' message
-func (api *API) UploadImageHandler(w http.ResponseWriter, req *http.Request) {
+// CreateDownloadHandler is a handler that adds a new download to an existing image
+func (api *API) CreateDownloadHandler(w http.ResponseWriter, req *http.Request) {
```

Comment:
Some of the handler functions are getting very long - consider breaking them down into smaller functions if it makes sense to.

Reply:
I agree with this in principal, but for this PR I'm trying to bring consistency with other endpoints and keep the code simple and easy to follow. The task I am moving on to next is titled "Tidy up existing image API endpoints to ensure they match the swagger spec". This will involve reviewing all the handlers as a whole. Any commonality would be easier to address under that PR. I'm also trying not to over-engineer things at this time given the limited time remaining to develop these.